File: Project-source/terrane-admin-web/src/lib/wizard.ts
/** 初始化向导 API（/admin-api/v1/wizard）— 仅超管可操作。
 *  步骤：License→超管→邮件→Branding→完成。配置写平台库 terrane_main。 */

import { request } from "@/lib/api";

export type StepStatus = "done" | "current" | "pending";

export interface WizardStep {
  key: "license" | "super_admin" | "email" | "branding";
  status: StepStatus;
}

export type Encryption = "auto" | "ssl" | "starttls" | "none";

export interface EmailState {
  configured: boolean;
  host: string;
  port: number;
  encryption: Encryption;
  username: string;
  from_address: string;
  from_name: string;
  allow_insecure: boolean;
  has_password: boolean;
}

export interface EmailPreset {
  id: string;
  label: string;
  host: string;
  port: number;
  encryption: string;
  from_locked: boolean;
  password_hint: string;
}

export interface BrandingState {
  product_name: string;
  logo_data: string | null;
  login_logo: string | null;
  favicon: string | null;
  accent_color: string;
  login_subtitle: string | null;
  support_url: string | null;
  enabled: boolean;
}

export interface WizardState {
  completed: boolean;
  steps: WizardStep[];
  email: EmailState;
  branding: BrandingState;
  email_presets: EmailPreset[];
}

export interface EmailConfigInput {
  host: string;
  port: number;
  encryption: Encryption;
  username: string;
  password: string;
  from_address: string;
  from_name: string;
  allow_insecure: boolean;
}

export interface BrandingInput {
  product_name: string;
  logo_data: string | null;
  login_logo?: string | null;
  favicon?: string | null;
  accent_color: string;
  login_subtitle: string | null;
  support_url: string | null;
}

const opts = (body?: unknown): RequestInit => ({
  method: "POST",
  credentials: "include",
  headers: { "Content-Type": "application/json" },
  ...(body ? { body: JSON.stringify(body) } : {}),
});

export function getWizard(): Promise<WizardState> {
  return request<WizardState>("/admin-api/v1/wizard", { credentials: "include" });
}

export function saveEmail(input: EmailConfigInput): Promise<unknown> {
  return request("/admin-api/v1/wizard/email", opts(input));
}

export function testEmail(to: string): Promise<unknown> {
  return request("/admin-api/v1/wizard/email/test", opts({ to }));
}

export function saveBranding(input: BrandingInput): Promise<unknown> {
  return request("/admin-api/v1/wizard/branding", opts(input));
}

export function completeWizard(): Promise<unknown> {
  return request("/admin-api/v1/wizard/complete", opts());
}

File: Project-source/terrane-admin-server/app/schemas/wizard.py
"""初始化向导 API schema（Pydantic v2 strict）。"""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, ConfigDict, EmailStr, Field

_strict = ConfigDict(strict=True, extra="forbid")

Encryption = Literal["auto", "ssl", "starttls", "none"]


class EmailConfigIn(BaseModel):
    model_config = _strict
    host: str = Field(min_length=1, max_length=255)
    port: int = Field(ge=1, le=65535)
    encryption: Encryption = "auto"
    username: str = Field(default="", max_length=255)
    password: str = Field(default="", max_length=512)
    from_address: EmailStr
    from_name: str = Field(default="Terrane", max_length=120)
    allow_insecure: bool = False  # 内网自签证书放行（默认严格校验）


class EmailTestIn(BaseModel):
    model_config = _strict
    to: EmailStr


class EmailPresetOut(BaseModel):
    id: str
    label: str
    host: str
    port: int
    encryption: str
    from_locked: bool
    password_hint: str


class BrandingIn(BaseModel):
    model_config = _strict
    product_name: str = Field(min_length=1, max_length=120)
    logo_data: str | None = Field(default=None, max_length=1_000_000)   # 控制台/工作区 Logo（data URI/URL）
    login_logo: str | None = Field(default=None, max_length=1_000_000)  # 登录页 Logo
    favicon: str | None = Field(default=None, max_length=1_000_000)     # 站点 favicon
    accent_color: str = Field(min_length=4, max_length=16)
    login_subtitle: str | None = Field(default=None, max_length=255)
    support_url: str | None = Field(default=None, max_length=512)


class StepOut(BaseModel):
    key: str
    status: str  # done | current | pending


class EmailStateOut(BaseModel):
    configured: bool
    host: str
    port: int
    encryption: str
    username: str
    from_address: str
    from_name: str
    allow_insecure: bool
    has_password: bool


class BrandingOut(BaseModel):
    product_name: str
    logo_data: str | None = None
    accent_color: str
    login_subtitle: str | None = None
    support_url: str | None = None
    enabled: bool


class WizardStateOut(BaseModel):
    completed: bool
    steps: list[StepOut]
    email: EmailStateOut
    branding: BrandingOut
    email_presets: list[EmailPresetOut]
